src/services/expenses.service.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const { v4: uuidv4 } = require('uuid');
+
+let expenses = [];
+
+const getAll = ({
+  userId,
+  categories,
+  from,
+  to,
+}) => {
+  let copy = [...expenses];
+
+  if (userId) {
+    copy = copy.filter(expense => expense.userId === userId);
+  }
+
+  if (categories) {
+    copy = copy.filter(expense => Array.isArray(categories)
+      ? categories.includes(expense.category)
+      : categories === expense.category
+    );
+  }
+
+  if (from) {
+    const fromDate = new Date(from);
+
+    copy = copy.filter(expense => new Date(expense.spentAt) >= fromDate);
+  }
+
+  if (to) {
+    const toDate = new Date(to);
+
+    copy = copy.filter(expense => new Date(expense.spentAt) <= toDate);
+  }
```

Comment:
so if you have all params, you will iterate 4 times?
better to do this logic due to 1 iteration only